apple/swift-format
# Generate rule docs automatically

## Summary 
- First step of #606, adds documentation for the list of available rules in `Documentation` directory.

## Changes
- Adds a `generate-pipeline` generator `RuleDocumentationGenerator` that extracts each rule DocC comment, and writes it as a table into `Documentation/RuleDocumentation.md`.
- In `Documentation/Configuration.md`, I've added a paragraph on rules configuration that links to the new auto-generated docs file, and also tells users that they can run `swift-format dump-configuration` to check out the list of available rules.

## How to review this
- Check out the format of the markdown file in `RuleDocumentation.md`.
- Navigate to Documentation files in the PR repo explorer. Does the markup and content look good?
- Does the layout make sense? I started with a table layout, but it wasn't as readable as I would hope (no way to tell lines to vertically align to top, no way to specify column width on GitHub markdown viewer — let's leave this until we can move to DocC?)

## TODOs

- [x] Rule description markup breaks where the original DocC content had a line break just because of long lines.
